paasta_tools/setup_kubernetes_job.py

```diff
@@ -102,19 +104,50 @@ def main() -> None:
 
     deploy_metrics = metrics_lib.get_metrics_interface("paasta")
 
-    # system_paasta_config = load_system_paasta_config()
     kube_client = KubeClient()
+    service_instances_valid = True
 
-    ensure_namespace(kube_client, namespace="paasta")
-    setup_kube_succeeded = setup_kube_deployments(
-        kube_client=kube_client,
-        service_instances=args.service_instance_list,
-        soa_dir=soa_dir,
+    # validate the service_instance names
+    service_instances_with_valid_names = get_service_instances_with_valid_names(
+        service_instances=args.service_instance_list
+    )
+
+    # returns a list of pairs of (error?, KubernetesDeploymentConfig) for every service_instance
+    service_instance_configs_list = get_kubernetes_deployment_config(
+        service_instances_with_valid_names=service_instances_with_valid_names,
         cluster=args.cluster or load_system_paasta_config().get_cluster(),
-        rate_limit=args.rate_limit,
-        metrics_interface=deploy_metrics,
+        soa_dir=soa_dir,
     )
-    sys.exit(0 if setup_kube_succeeded else 1)
+    if ((False, None) in service_instance_configs_list) or (
+        len(service_instances_with_valid_names) != len(args.service_instance_list)
+    ):
+        service_instances_valid = False
```

Comment:
Oops, I'm realizing you already have error logging in `get_kubernetes_deployment_config` which makes this log.error line (and the one on line 117) redundant. Feel free to remove these. The errors from get_kubernetes_deployment_config are a little more informative since they include the problematic service/instance.